dicoogle/src/main/java/pt/ua/dicoogle/server/web/servlets/mlprovider/TrainServlet.java

```diff
@@ -0,0 +1,100 @@
+
+    private static final Logger log = LoggerFactory.getLogger(TrainServlet.class);
+
+    @Override
+    protected void doPost(HttpServletRequest req, HttpServletResponse resp) throws ServletException, IOException {
+
+        String dataString = IOUtils.toString(req.getReader());
+        if (dataString == null) {
+            resp.sendError(404, "Empty POST body");
+            return;
+        }
+
+        ObjectMapper mapper = new ObjectMapper();
+        TrainRequest trainRequest;
+        try {
+            trainRequest = mapper.readValue(dataString, TrainRequest.class);
+            MLProviderInterface mlplugin = PluginController.getInstance().getMachineLearningProviderByName(trainRequest.getProvider(), true);
+            if(mlplugin == null){
+                log.error("A provider with the provided name does not exist");
+                resp.sendError(404, "Malformed request");
+            } else {
+                MLTrainTask trainTask = mlplugin.trainModel(trainRequest.getModelID());
+                switch (trainTask.getStatus()){
+                    case BUSY:
+                        log.error("Could not create training task, service is busy");
+                        resp.sendError(405, "Could not create training task, service is busy");
+                        break;
+                    case REJECTED:
+                        log.error("Could not create training task, request is malformed");
+                        resp.sendError(404, "Could not create training task, service is busy");
+                        break;
+                    default:
+                        resp.setContentType("application/json");
+                        PrintWriter out = resp.getWriter();
+                        mapper.writeValue(out, trainTask);
+                        out.close();
+                        out.flush();
+                }
+            }
+        } catch (Exception e) {
+            log.error("Error parsing json string", e);
+            resp.sendError(404, "Malformed request");
+        }
+    }
+
+    @Override
+    protected void doDelete(HttpServletRequest req, HttpServletResponse resp) throws ServletException, IOException {
+
+        String dataString = IOUtils.toString(req.getReader());
+        if (dataString == null) {
+            resp.sendError(404, "Empty POST body");
+            return;
+        }
+
+        ObjectMapper mapper = new ObjectMapper();
+        TrainRequest trainRequest;
+        try {
+            trainRequest = mapper.readValue(dataString, TrainRequest.class);
+            MLProviderInterface mlplugin = PluginController.getInstance().getMachineLearningProviderByName(trainRequest.getProvider(), true);
+            if(mlplugin == null){
+                log.error("A provider with the provided name does not exist");
+                resp.sendError(404, "Malformed request");
+            } else {
+                boolean stopped = mlplugin.stopTraining("");
```

Comment:
This is semi-unfinished, maybe I will mark it as not implemented. The API expects a training task ID, because I'm assuming providers can run several training tasks at once. MONAI however does not support multiple training tasks, at least with default configurations, and this is why this task ID is empty.